content/departments/cloud/technical-docs/operations.md

```diff
@@ -432,6 +432,59 @@ resource.type="gce_instance"
 protoPayload.authenticationInfo.principalEmail="system@google.com"
 ```
 
+### Enabling GCP Cloud Tracing on Managed Instances
+
+As part of the [Centralised Observability](https://github.com/sourcegraph/customer/issues/1151) efforts we will be enabling GCP Cloud Tracing on all Managed Instances.
+
+To enable GCP Cloud Tracing on an existing instance, follow these instructions for deploying the `opentelemetry-collector` service:
+
+1. `cd` into the desired customer directory and into the deployment folder (`red` or `black`).
+2. Create a new directory called `otel-collector` and within it a folder called `config.yaml` with the following contents:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+
+exporters:
+  jaeger:
+    endpoint: '$JAEGER_HOST:14250'
+    tls:
+      insecure: true
+  googlecloud:
+    retry_on_failure:
+      enabled: false
+
+extensions:
+  health_check:
+    port: 13133
+  zpages:
+    endpoint: 'localhost:55679'
+
+service:
+  extensions: [health_check, zpages]
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [jaeger, googlecloud]
+```
+
+3. Update the corresponding `docker-compose.override.yaml` file to now include the `otel-collector` service and configure all existing services with the correct environment variables. Reference this [pull request enabling otel for `tpgi`](https://github.com/sourcegraph/deploy-sourcegraph-managed/pull/911/files) for an example of all that needs to be modified.
+4. Connect to the manage instance and add the following configuration to the `site-config`:
+
+```json
+
+"observability.tracing": {
+  "type": "opentelemetry",
+  "sampling": "selective",
+  "urlTemplate": "https://google.com?{{.TraceID}}"
```

Comment:
🤯 this URL template just works?

Reply:
As far as I can tell, this field doesn't actually seem to be used with the `googleCloud` exporter. It must have the correct URL endpoints correctly configured. Traces with this configuration were correctly delivered.

Reply:
The URL template is for links _to_ the traces rendered in the UI, as well as occasionally for logs

Reply:
Ah. Then no, this URL will almost certainly not work. 🤣 I'll look into what the correct URL should be.

Reply:
Example: https://console.cloud.google.com/traces/list?project=sourcegraph-managed-sg&tid=166dd657862e1dda79009d5f0c78e523

If this works I think the template should be something like:

```
https://console.cloud.google.com/traces/list?referrer=search&project=sourcegraph-managed-sg&tid={{.TraceID}}
```